integrations/langchain/src/unitycatalog_ai_langchain/toolkit.py

```diff
@@ -0,0 +1,135 @@
+import json
+import logging
+import os
+from typing import Any, Dict, List, Optional
+
+from langchain_core.pydantic_v1 import BaseModel, Field, root_validator
+from langchain_core.tools import StructuredTool, Tool
+from unitycatalog.ai.client import BaseFunctionClient, get_uc_function_client
+from unitycatalog.ai.utils import (
+    generate_function_input_params_schema,
+    validate_full_function_name,
+)
+
+_logger = logging.getLogger(__name__)
+
+UC_LIST_FUNCTIONS_MAX_RESULTS = 100
+
+
+def get_tool_name(func_name: str) -> str:
+    if len(func_name) > 64:
+        tool_name = func_name[-64:]
+        _logger.warning(
+            f"Function name {func_name} is too long, truncating to 64 characters {tool_name}."
+        )
+        return tool_name
+    return func_name
+
+
+def validate_or_set_default_client(client: Optional[BaseFunctionClient] = None):
+    client = client or get_uc_function_client()
+    if client is None:
+        raise ValueError(
+            "No client provided, either set the client when creating the "
+            "LangchainToolkit, or set the default client using "
+            "unitycatalog.ai.client.set_uc_function_client(client)."
+        )
+    return client
+
+
+class LangchainToolkit(BaseModel):
+    function_names: List[str] = Field(
+        default_factory=list,
+        description="The list of function names in the form of 'catalog.schema.function'",
+    )
+
+    tools_dict: Dict[str, Tool] = Field(default_factory=dict)
+
+    client: Optional[BaseFunctionClient] = Field(
+        default=None,
+        description="The client for managing functions, must be an instance of BaseFunctionClient",
+    )
+
+    class Config:
+        arbitrary_types_allowed = True
+
+    @root_validator
+    def validate_toolkit(cls, values) -> Dict[str, Any]:
+        client = validate_or_set_default_client(values.get("client"))
+        values["client"] = client
+
+        function_names = values["function_names"]
+        tools_dict = values["tools_dict"]
+        for name in function_names:
+            if name not in tools_dict:
+                full_func_name = validate_full_function_name(name)
+                if full_func_name.function_name == "*":
+                    token = "None"
```

Comment:
```suggestion
                    token = None
```

why is this a string?

Reply:
So that I can enter the while loop `while token is not None`

Reply:
Let's modify the code. `token = "None"` does look like an error.